_posts/2016-09-16-why-linting-errors-should-cause-tests-to-fail.md

```diff
@@ -0,0 +1,138 @@
+a hotfix. For example:
+
+* Transient test failures cause by not properly mocking out the system clock
+* Test failures introduced by your package repository being down.
+
+I don't think anyone would suggest disabling tests all together just to avoid
+the possibility that one of these errors might block a hotfix.
+
+If you don't have a system that can work around those types of failures, you
+properly should. And if you _do_ have a system that can work around these
+failures, it would be appropriate to use it to work around a hypothetical
+misaligned curly brace.
+
+## Why I care so much
+
+By putting test failures on a pedestal above linting errors, I think we
+underestimate the value of linting, and the harm that linting errors can cause.
+
+Without enforcing your linting rules as part of your tests, you are guaranteed
+to end up with linting errors merged into your master branch. Let's explore
+three phases of a developer's workflow and see how they are negatively affected
+by this choice.
+
+### During code review
+
+If a reviewer can operate under the assumption that the code they are reviewing
+conforms to the project's linting rules, they don't need to manually lint the
+code they are reading.
+
+For example, if the reviewer sees that the author has removed a variable
+definition, they can trust that the author did not overlook a later usage of
+that variable. Without the linter's assurance, the reviewer would need to
+manually check the entire file for other usages of that variable.
+
+This applies to less overt errors as well. Did the author use the right number
+of spaces when they indented? If tests were configured to fail on linting
+errors, the reviewer wouldn't even need to ask herself that question, freeing
+her mental bandwidth to focus on other, more substantial, questions.
+
+### During deploy
+
+In addition to testing for issues of style, linters check for code validity. We
+should not deploy code that is invalid.
+
+For example, calling `console.log` will cause your application to crash in
+Internet Explorer 9. Despite this dire consequences, developers frequently
+open pull requests with `console.log` statements leftover from debugging.
+A linting rule like [`no-console`](http://eslint.org/docs/rules/no-console), if
+enforced as part of your test suite, can give you complete confidence that you
+will never encounter that particular bug in production.
+
+### On a new feature branch
+
+When a developer starts a new feature branch, they start their branch off of
+master. This means they are starting their work on a branch that could already
+have linting errors. When it's time for her to ship her changes she is now
+forced to choose between cleaning up somebody else's mess, or leaving these
+linting errors in master.
+
+The presence of these errors also reinforces a mentality that such errors are
+acceptable in master. While a single linting failure may seem quite harmless,
```

Comment:
I'm deleting this comment, which I should have done before "submitting" the review in the first place. Thanks New GitHub Feedback Flow!